docs/developer-env.md

```diff
@@ -0,0 +1,530 @@
+
+WSL is an optional feature on Windows 10 version 2004 and higher (Build 19041 and higher) or Windows 11. You can enable it through PowerShell or Windows Command Prompt. The steps below outline the recommended setup.
+
+<!---
+### Windows Features dialog
+
+To enable WSL on Windows using Windows Features dialog:
+
+1. In the Windows search bar, enter 'features' to bring up the **Turn Windows Features on and off** dialog.
+1. Scroll down and check **Windows Subsystem for Linux**.
+1. Select **OK** and restart Windows.
+1. After restarting Windows, check that you have WSL enabled by opening a Command Prompt or PowerShell and typing `wsl`.
+
+To install Ubuntu on WSL:
+
+1. Go to **Start Button > Microsoft Store**.
+1. Enter 'Linux' into the search field, then click **Run Linux on Windows**.
+1. Select the latest Ubuntu distribution.
+1. Select **Get** and wait for Windows to download and install Ubuntu.
+1. When it’s finished, select **Launch**.
+1. A terminal window will appear. Wait for Ubuntu to finish installing, then create a new Linux username and password when prompted.
+
+### PowerShell or Windows Command Prompt
+--->
+
+To enable WSL on Windows using Powershell or Windows Command Prompt:
+
+1. Open PowerShell or Windows Command Prompt in administrator mode by right-clicking and selecting **Run as administrator**.
+1. Run `wsl --install`.
+
+    :::{note}
+    This command will enable the features necessary to run WSL and install the Ubuntu distribution.
+    :::
+
+1. When prompted, restart Windows.
+1. After restarting Windows, open the Ubuntu distribution using the **Start** menu and create a new Linux **User Name** and **Password** when prompted.
+
+    :::{note}
+    The **User Name** and **Password** is specific to each Linux distribution that you install and has no bearing on your Windows user name.
+    :::
+
+See [Set up a WSL development environment](https://learn.microsoft.com/en-us/windows/wsl/setup/environment) for more information.
+
+## Development Containers
+
+[Development Containers](https://containers.dev/) (Dev Containers), are Docker containers that are specifically configured to provide a fully featured development environment. They can be used to run an application, to separate tools, libraries, or runtimes needed for working with a codebase. Dev Containers can be run locally or remotely, in a private or public cloud, and in a variety of supporting tools and editors.
+
+VS code and Docker are required to create and manage your Dev Containers. See {ref}`vscode-install` and {ref}`docker-desktop` for installation instructions.
+
+### Development Containers extension
+
+The VS Code Dev Containers extension lets you use a container as a full-featured development environment. It allows you to open any folder inside (or mounted into) a container and take advantage of VS Code's full feature set.
```

Comment:
We've said "full-featured" 3 times in the past few paragraphs. Maybe we can remove these two or use a different adjective? I'm coming up blank right now though!